### PR TITLE
feat(shadow): add world-space CSM bias

### PR DESCRIPTION
**Pull request by deltakosh**
## Summary
- add optional `worldSpaceBias` for CSM caster depth offsets authored in world units
- reserve matching far-plane headroom so tightly fitted far casters remain inside the clip volume
- preserve legacy normalized `bias` behavior when `worldSpaceBias` is omitted
- document the projection math and add focused edge-case coverage

## Validation
- `pnpm run lint`
- `pnpm test` completed with all bundle ceilings and 426 parity cases passing; only the four known nondeterministic physics scenes 47, 104, 105, and 106 failed locally
- visually checked scene 214 with legacy and world-space bias paths

### CODE REVIEW

**Comment by bjsplat** [comment]
## API Changes

API Extractor detected public API changes for `@babylonjs/lite`.

No removed public API lines were detected; this appears to be additive.

<details>
<summary>API Extractor diff</summary>

```diff
diff --git a/home/vsts/work/1/s/test-results/api-report/target/temp/babylon-lite.api.md b/home/vsts/work/1/s/test-results/api-report/current/temp/babylon-lite.api.md
index 95d56a9b..dddb97ab 100644
--- a/home/vsts/work/1/s/test-results/api-report/target/temp/babylon-lite.api.md
+++ b/home/vsts/work/1/s/test-results/api-report/current/temp/babylon-lite.api.md
@@ -1711,6 +1711,7 @@ export interface CsmDirectionalShadowGeneratorConfig {
     numCascades?: number;
     shadowMaxZ?: number;
     stabilizeCascades?: boolean;
+    worldSpaceBias?: number;
 }
 
 // @public
```

</details>

**Comment by bjsplat** [comment]
### Lab - Static Site

[Open deployed site](https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/lite/20260711.1/lab/index.html)

<sub>Build 20260711.1 - merge @ d513a2451f7b08ebd18c5d88efef771dcc2f1544</sub>

**Comment by bjsplat** [comment]
### 📋 perf — Test Report

[View full Playwright report](https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/lite/20260711.1/testResults/perf/index.html)

<sub>Build 20260711.1 · merge @ d513a2451f7b08ebd18c5d88efef771dcc2f1544</sub>

**Comment by bjsplat** [comment]
## API Changes

API Extractor detected public API changes for `@babylonjs/lite`.

No removed public API lines were detected; this appears to be additive.

<details>
<summary>API Extractor diff</summary>

```diff
diff --git a/home/vsts/work/1/s/test-results/api-report/target/temp/babylon-lite.api.md b/home/vsts/work/1/s/test-results/api-report/current/temp/babylon-lite.api.md
index 95d56a9b..dddb97ab 100644
--- a/home/vsts/work/1/s/test-results/api-report/target/temp/babylon-lite.api.md
+++ b/home/vsts/work/1/s/test-results/api-report/current/temp/babylon-lite.api.md
@@ -1711,6 +1711,7 @@ export interface CsmDirectionalShadowGeneratorConfig {
     numCascades?: number;
     shadowMaxZ?: number;
     stabilizeCascades?: boolean;
+    worldSpaceBias?: number;
 }
 
 // @public
```

</details>

**Comment by bjsplat** [comment]
### Lab - Static Site

[Open deployed site](https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/lite/20260711.2/lab/index.html)

<sub>Build 20260711.2 - merge @ 2b7189e0e834b1cff01481f710a420b5a07fe6dc</sub>

**Comment by bjsplat** [comment]
### 📋 perf — Test Report

[View full Playwright report](https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/lite/20260711.2/testResults/perf/index.html)

<sub>Build 20260711.2 · merge @ 2b7189e0e834b1cff01481f710a420b5a07fe6dc</sub>

**Comment by bjsplat** [comment]
### 📋 parity — Test Report

[View full Playwright report](https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/lite/20260711.2/testResults/parity/index.html)

<sub>Build 20260711.2 · merge @ 2b7189e0e834b1cff01481f710a420b5a07fe6dc</sub>

**Comment by bjsplat** [comment]
## API Changes

API Extractor detected public API changes for `@babylonjs/lite`.

No removed public API lines were detected; this appears to be additive.

<details>
<summary>API Extractor diff</summary>

```diff
diff --git a/home/vsts/work/1/s/test-results/api-report/target/temp/babylon-lite.api.md b/home/vsts/work/1/s/test-results/api-report/current/temp/babylon-lite.api.md
index 95d56a9b..dddb97ab 100644
--- a/home/vsts/work/1/s/test-results/api-report/target/temp/babylon-lite.api.md
+++ b/home/vsts/work/1/s/test-results/api-report/current/temp/babylon-lite.api.md
@@ -1711,6 +1711,7 @@ export interface CsmDirectionalShadowGeneratorConfig {
     numCascades?: number;
     shadowMaxZ?: number;
     stabilizeCascades?: boolean;
+    worldSpaceBias?: number;
 }
 
 // @public
```

</details>

**Comment by bjsplat** [comment]
### Lab - Static Site

[Open deployed site](https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/lite/20260711.3/lab/index.html)

<sub>Build 20260711.3 - merge @ 16063ebb4d2b53059fd3a6686e1a56278c13feb6</sub>
